power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -885,49 +882,52 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
         return run_functor_with_all_types_return_vector(check_each_component);
     }
 
-    std::vector<std::pair<std::string, bool>> is_update_independent(ConstDataset const& update_data) {
-        std::vector<std::pair<std::string, bool>> result;
+    std::unordered_map<std::string, bool> is_update_independent(ConstDataset const& update_data) {
+        std::unordered_map<std::string, bool> result;
 
         // If the batch size is (0 or) 1, then the update data for this component is 'independent'
         if (update_data.batch_size() <= 1) {
-            result.emplace_back("all component", true);
+            result["all component"] = true;
             return result;
         }
 
         auto const all_comp_update_independence = check_components_independence(update_data);
         for (auto const& comp : all_comp_update_independence) {
-            result.emplace_back(comp.name, comp.is_independent());
+            result[comp.name] = comp.is_independent();
         }
 
         return result;
     }
+
     void validate_update_data_independence(UpdateCompProperties const& comp) const {
-        if (!comp.has_id && comp.ids_all_na) {
+        if (comp.is_empty_component()) {
             return; // empty dataset is still supported
         }
         if (comp.elements_in_base < comp.elements_ps_in_update) {
-            throw DatasetError("Update data has more elements per scenario than input data for component " + comp.name +
-                               "!");
+            throw DatasetError(
+                "Dataset error: Update data has more elements per scenario than input data for component " + comp.name +
```

Comment:
here and below: `DatasetError` already prepends `Dataset error:` so it already should decode correctly as-is. Instead, now it will log `Dataset error: Dataset error: xyz`, which is redundant.

```cpp
class DatasetError : public PowerGridError {
  public:
    explicit DatasetError(std::string const& msg) { append_msg("Dataset error: " + msg); }
};
```